### ISSUE TITLE
PHP 8 / PSR12: Variable-length union arguments generate invalid OperatorSpacing errors

### ISSUE THREAD

**Issue by zimzat** [squizlabs/PHP_CodeSniffer#3267]
**Describe the bug**
Variable-length union arguments generate invalid "expected at least one space" error. 

**Code sample**
```php
<?php

function test(string|int ...$values)
{
}
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs -s test.php --standard=PSR12`
3. See error message displayed
```
FILE: test.php
---------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
---------------------------------------------------------------------------------------------------------------
 3 | ERROR | [x] Expected at least 1 space before "|"; 0 found
   |       |     (PSR12.Operators.OperatorSpacing.NoSpaceBefore)
 3 | ERROR | [x] Expected at least 1 space after "|"; 0 found
   |       |     (PSR12.Operators.OperatorSpacing.NoSpaceAfter)
---------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
---------------------------------------------------------------------------------------------------------------

Time: 41ms; Memory: 6MB
```

**Expected behavior**
The code is correct as-is and should not be flagged as an error to be fixed.

**Versions (please complete the following information):**
 - OS: Linux Ubuntu 18.04.1
 - PHP: 8.0.3
 - PHPCS: dev-master 46715ed9eb3842096581a9d88c6b766696951b19
 - Standard: PSR12

**Additional context**
Removing the `...` makes the error go away. Running `phpcbf` changes the code to `(string | int ...$values)` and `phpcs` stops warning of an error.

**Comment by jrfnl** [comment]
@zimzat Thanks for reporting this. I'm looking into this and expect to pull a fix later today.

**Comment by jrfnl** [comment]
@zimzat PR #3268 should fix this. Testing appreciated.

**Comment by zimzat** [comment]
@jrfnl We only have one place that scenario occurred (fwiw) and it's working as expected with that PR, thanks! :+1:

**Comment by jrfnl** [comment]
@zimzat Thanks for testing!
